### PR TITLE
[SPARK-23336][BUILD] Upgrade snappy-java to 1.1.7.1

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

This PR upgrade snappy-java from 1.1.2.6 to 1.1.7.1. 
1.1.7.1 release notes: 
- Improved performance for big-endian architecture
- The other performance improvement in [snappy-1.1.5](https://github.com/google/snappy/releases/tag/1.1.5)

1.1.4 release notes: 
- Fix a 1% performance regression when snappy is used in PIE executables.
- Improve compression performance by 5%.
- Improve decompression performance by 20%.

More details:
https://github.com/xerial/snappy-java/blob/master/Milestone.md

## How was this patch tested?

manual tests

### CODE REVIEW

**Review by smurakozi — COMMENTED** [review]
I think this change is ok.

**Comment by wangyum** [comment]
Retest this please.

File: dev/deps/spark-deps-hadoop-2.6 [hunk 1]
```diff
@@ -182,7 +182,7 @@ slf4j-api-1.7.16.jar
 slf4j-log4j12-1.7.16.jar
 snakeyaml-1.15.jar
 snappy-0.2.jar
-snappy-java-1.1.2.6.jar
+snappy-java-1.1.4.jar
```
Comment (dongjoon-hyun) [inline dev/deps/spark-deps-hadoop-2.6]: Is there any reason to use `1.1.4` instead of `1.1.7.1`?

Comment (kiszk) [reply dev/deps/spark-deps-hadoop-2.6]: +1

**Comment by srowen** [comment]
This looks like a safe upgrade, though it's always a little risky because components like Hadoop use this too, and not the same version. Tests will help show whether there's an immediate issue.

**Comment by wangyum** [comment]
Retest this please.

**Comment by wangyum** [comment]
The failure is due to flaky test suite.
```
org.apache.spark.sql.hive.client.HiveClientSuites.(It is not a test it is a sbt.testing.NestedSuiteSelector)
```

jenkins, retest this please.

**Comment by wangyum** [comment]
jenkins, retest this please.

**Comment by wangyum** [comment]
jenkins, retest this please.

**Review by felixcheung — APPROVED** [review]
LG, same concerns as Sean

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**